### PR TITLE
rspec-branch

### PR DESCRIPTION
**Pull request by m15e**
new branch because workflows isn't working for some reason

- [ ] Rspec tests for Enumerable methods

### CODE REVIEW

File: spec/enumerables_spec.rb [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  # my_each
+  describe '#my_each' do
+    it 'returns the same value as the original given an array and a block' do
+      expect(spec_num_array.my_each(&:even?)).to eql(test_array)
+    end
+
+    it 'returns the same value as the original given a range and a block' do
+      expect(spec_range.my_each(&:even?)).to eql(test_range)
+    end
+  end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb]: Check if it's an Enumerator

File: spec/enumerables_spec.rb [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  # my_each_with_index
+  describe '#my_each_with_index' do
+    it 'returns the same value as the original given an array and a block' do
+      expect(spec_num_array.my_each_with_index { |a, b| a + b }).to eql(spec_num_array.each_with_index { |a, b| a + b })
+    end
+
+    it 'returns the same value as the original given a range and a block' do
+      expect(spec_range.my_each_with_index { |a, b| a + b }).to eql(spec_range.each_with_index { |a, b| a + b })
+    end
+  end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb]: Check if it's an Enumerator

File: spec/enumerables_spec.rb [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    end
+
+    it 'returns the same value as the original given a range and a block' do
+      expect(spec_range.my_select(&:even?)).to eql(spec_range.select(&:even?))
+    end
+
+    it 'returns the same value as the original given a hash and a block' do
+      expect(spec_hash.my_select { |_k, v| v.even? }).to eql(spec_hash.select { |_k, v| v.even? })
+    end
+  end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb]: Check if it's an Enumerator

File: spec/enumerables_spec.rb [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    end
+
+    it 'returns true if one value in a range is true.' do
+      expect(spec_range.my_any?(2)).to eql(true)
+    end
+
+    it 'returns true if one key in the hash is true.' do
+      expect(spec_hash.my_any? { |key, _val| key == :a }).to eql(true)
+    end
+  end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb]: Check for pattern, String or Integer, ...

File: spec/enumerables_spec.rb [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  # my_all
+  describe 'my_all' do
+    it 'returns true if all values are smaller than 21' do
+      expect(spec_num_array.my_all? { |v| v < 21 }).to eql(spec_num_array.all? { |v| v < 21 })
+    end
+
+    it 'returns false if a value in the array doesn\'t satisfy the block condition' do
+      expect(spec_num_array.my_all? { |v| v < 10 }).to eql(spec_num_array.all? { |v| v < 10 })
+    end
+  end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb]: What about testing if all the array contains only one value

File: spec/enumerables_spec.rb [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+  # my_any
+  describe '#my_any' do
+    it 'returns true if one value in the array is true.' do
+      expect(spec_num_array.my_any?(2)).to eql(true)
+    end
+
+    it 'returns true if one value in a range is true.' do
+      expect(spec_range.my_any?(2)).to eql(true)
+    end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb]: Also, add test for empty or falsy array

File: spec/enumerables_spec.rb [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+    it 'behaves like the original when applied to a range' do
+      expect(spec_range.my_inject { |sum, val| sum + val }).to eql(spec_range.inject { |sum, val| sum + val })
+    end
+
+    it 'finds the longest string' do
+      expect(word_array.my_inject { |a, b| a.length > b.length ? a : b }).to eql('very_long_string')
+    end
+  end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb]: Write for operation like `:+` and/or `:*`

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for that first step into RSpec.
However, there are some changes to need to be done

1. I got this error when I ran `rspec`
```rb
$ rspec spec/enumerables_spec.rb 

An error occurred while loading ./spec/enumerables_spec.rb. - Did you mean?
                    rspec ./spec/spec_helper.rb

Failure/Error: require 'enumerable'

LoadError:
  cannot load such file -- enumerable
# ./spec/enumerables_spec.rb:1:in `<top (required)>'
No examples found.


Finished in 0.00004 seconds (files took 0.48227 seconds to load)
0 examples, 0 failures, 1 error occurred outside of examples
```

2. Read the comments left below

Kindly make the changes and submit a new code review request.
Thanks

Mael FOSSO

**Review by mcrd25 — APPROVED** [review]
# Changes Approved :heavy_check_mark: 


Hi Team! :wave: 

Good job with changers, all test pass! :+1: 



***If you have any questions, now or in the future, do not hesitate to contact me on social media, email or on slack and I will be glad to chat with you.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)
